### DISCUSSION TITLE
Arista not pulling bgp peers with local address as interface

### DISCUSSION THREAD

**Discussion by lovkeshw131** [napalm-automation/napalm#1859] [General]
I adding this in my local repo and it worked and tried to create a PR but looks like i don't have permssions. Can you guys please look into it and add it or modify it

            if group_or_neighbor.startswith('interface'):
                # Arista supports defining interface names as a peer for v6
                # It used link local address to establish such peers
                # eg: neighbor interface Po20 peer-group XYZ-V6 remote-as XXX
                peer_address = group_or_neighbor + ' ' + options[0]
                # It will pick up interface and interface name, eg: interface Po20
                options = options[1:]
                if peer_address not in bgp_neighbors:
                    bgp_neighbors[peer_address] = default_neighbor_dict(local_as)
                if options[0] == "peer-group":
                    bgp_neighbors[peer_address]["__group"] = options[1]
                elif options[0] == "peer" and options[1] == "group":
                    bgp_neighbors[peer_address]["__group"] = options[2]
                bgp_neighbors[peer_address].update(
                    parse_options(options, default_value)
                )
            else:
                try:
                    # will try to parse the neighbor name
                    # which sometimes is the IP Address of the neigbor
                    # or the name of the BGP group
                    ipaddress.ip_address(group_or_neighbor)
                    # if passes the test => it is an IP Address, thus a Neighbor!
                    peer_address = group_or_neighbor

**Comment by bewing** [comment]
Possibly related:  https://github.com/napalm-automation/napalm/issues/1759

Can you be a bit more clear on the problem here?

* What getter are you trying to run and running into an issue?
* What traceback (if any?)
* if we are failing on parsing an output, do you know / can you provide what the router is sending to NAPALM?

**Reply by lovkeshw131** [reply]
device_driver = napalm.get_network_driver("eos")
>>> device = device_driver(hostname="xxx", username="xx.x", password="xxx", optional_args={'transport': 'ssh'})
a=device.get_bgp_config()

>>> a.keys()
dict_keys(['abc', 'interface'])


In above, instead of adding interface number in an existing bgp group, it added a new group interface

Actual configuration is like below:

**neighbor interface Po20 peer-group XYZ-V6 remote-as XXX**

It considers interface as a group name in above and creates a peer group there without any other config
